rllib/examples/inference/policy_inference_after_training_w_connector.py

```diff
@@ -0,0 +1,265 @@
+    # here).
+    print("Env-to-module ConnectorV2 ...")
+    env_to_module = EnvToModulePipeline(
+        input_observation_space=env.observation_space,
+        input_action_space=env.action_space,
+        connectors=[
+            AddObservationsFromEpisodesToBatch(),
+            AddStatesFromEpisodesToBatch(),
+            BatchIndividualItems(),
+            NumpyToTensor(),
+        ],
+    )
+
+    # Create the RLModule.
+    # Get the last checkpoint from the above training run.
+    best_result = results.get_best_result(
+        metric=f"{ENV_RUNNER_RESULTS}/{EPISODE_RETURN_MEAN}", mode="max"
+    )
+    # Create new Algorithm and restore its state from the last checkpoint.
+    rl_module = RLModule.from_checkpoint(
+        os.path.join(
+            best_result.checkpoint.path,
+            "learner",
+            "module_state",
+            DEFAULT_MODULE_ID,
+        )
+    )
+    print("RLModule restored ...")
+
+    # For the module-to-env pipeline, we will use the convenient config utility.
+    print("Module-to-env ConnectorV2 ...")
+    module_to_env = base_config.build_module_to_env_connector(env=env)
+
+    # Now our setup is complete:
+    # [gym.Env] -> env-to-module -> [RLModule] -> module-to-env -> [gym.Env] ... repeat
+    num_episodes = 0
+
+    obs, _ = env.reset()
+    episode = SingleAgentEpisode(
+        observations=[obs],
+        observation_space=env.observation_space,
+        action_space=env.action_space,
+    )
+
+    while num_episodes < args.num_episodes_during_inference:
+        shared_data = {}
+        input_dict = env_to_module(
+            episodes=[episode],  # ConnectorV2 pipelines operate on lists of episodes.
+            rl_module=rl_module,
+            explore=args.explore_during_inference,
+            shared_data=shared_data,
+        )
+        # No exploration.
+        if not args.explore_during_inference:
+            rl_module_out = rl_module.forward_inference(input_dict)
+        # Using exploration.
+        else:
+            rl_module_out = rl_module.forward_exploration(input_dict)
+
+        to_env = module_to_env(
+            data=rl_module_out,
+            episodes=[episode],  # ConnectorV2 pipelines operate on lists of episodes.
+            rl_module=rl_module,
+            explore=args.explore_during_inference,
+            shared_data=shared_data,
+        )
+        # Send the computed action to the env. Note that the RLModule and the
+        # connector pipelines work on batched data (B=1 in this case), whereas the Env
+        # is not vectorized here, so we need to use `action[0]`.
+        action = to_env.pop(Columns.ACTIONS)[0]
+        obs, reward, terminated, truncated, _ = env.step(action)
+        episode.add_env_step(
+            obs,
+            action,
+            reward,
+            terminated=terminated,
+            truncated=truncated,
+            # Same here: [0] b/c RLModule output is batched (w/ B=1).
```

Comment:
We should give here also the hint that we use the episode here to store the states of the module such that it can be used in the connectors and module without intervention.

Reply:
The user might otherwise question why to store all this and harm performance with it

Reply:
True! I agree, the next thing should be an "on-the-edge" example:
* RLModule (as ONNX re-import?)
* stateful ConnectorV2 (loaded from checkpoint?)
* Episode (how do we make this efficient and not carry around a maybe 10k ts episode?) <- here, I have to say though, extending an episode is just adding items to a list, which does NOT pose a large performance penalty
* what if all the above must run on non-python, e.g. C++ ??